src/packet/secret_key_macro.rs

```diff
@@ -150,6 +150,50 @@ macro_rules! impl_secret_key {
                     .build()?
                     .sign_key(key, key_pw, &self)
             }
+
+            pub fn set_password<O, N, R>(
+                &mut self,
+                pw_old: O,
+                pw_new: N,
+                rng: R,
+            ) -> $crate::errors::Result<()>
+            where
+                O: FnOnce() -> String,
+                N: FnOnce() -> Option<String>,
+                R: rand::Rng + rand::CryptoRng,
+            {
+                let s2k = $crate::types::S2kParams::new_default(rng);
+                Self::set_password_and_s2k(self, pw_old, pw_new, s2k)
+            }
+
+            pub fn set_password_and_s2k<O, N>(
+                &mut self,
+                pw_old: O,
+                pw_new: N,
+                s2k_params: $crate::types::S2kParams,
+            ) -> $crate::errors::Result<()>
+            where
+                O: FnOnce() -> String,
+                N: FnOnce() -> Option<String>,
+            {
+                use $crate::types::SecretParams;
+
+                let plain = match &self.secret_params {
+                    SecretParams::Plain(plain) => plain.clone(),
```

Comment:
would be nice to avoid this clone (but not critical)

Reply:
The clone also bugged me, but I can't think of a DRY way to avoid it.
It's of course avoidable, but I'm not sure that it's worth avoiding at the cost of messier code.

If you have an idea for a neat approach, I'd be more than happy to change it.

Reply:
I just realized that `PlainSecretParams::encrypt` (which I call, here) takes a `self` too. Which I don't think is necessary, a `&self` should be enough, there.

But unfortunately that makes my (mostly silly) cloning here more necessary.

Reply:
ok :/ lets leave it for now